### DISCUSSION TITLE
How can I display json data in a table

### DISCUSSION THREAD

**Discussion by yablochnyi** [filamentphp/filament#11140] [Help]
### Package

Table builder

### Package Version

v3

### How can we help you?

Hi, there is a market_position field in the table. It stores json
[{"symbol":"BTCUSDT","side":"long\/short","price":"40000.00","lots":"1.00","notional":"40000.00","unpl":"1.00"}]
How can I display symbol in the table?
`public function table(Table $table): Table
    {

        return $table
            ->query(UserStatistic::query()->where('account', $this->statistic->serial_number_user))
            ->columns([
                TextColumn::make('market_positions.symbol')
                    ->label('Symbol')

            ]);
    }`

**Comment by leandrocfe** [comment]
If you are using `market_positions.symbol`, the column value should be

```json
{
    "symbol": "BTCUSDT",
    "side": "long/short",
    "price": "40000.00",
    "lots": "1.00",
    "notional": "40000.00",
    "unpl": "1.00"
}
```

without "[]"

**Comment by Max-Lovell** [comment]
could do a custom view saved in resources/views/filament/tables/columns/json.blade.php:

```
@foreach($getState() as $variable => $value)
    <p>
        {{$variable}}={{$value}}@if(!$loop->last),@endif&nbsp;
    </p>
@endforeach
```

and call it in the filament page

```
ViewColumn::make('json data col')
    ->view('filament.tables.columns.json')
    ->searchable()
    ->sortable(),
```

**Reply by Benyaminrmb** [reply]
its good but how can we add toggledHiddenByDefault() in this ViewColumn?
cuz its not work

**Reply by Benyaminrmb** [reply]
`->toggleable(isToggledHiddenByDefault: true)
` my bad .  ty for `ViewColumn`

**Reply by AoukouKwadjo** [reply]
JSON value: 
{"cle1": "valeur1"}

I use this code for the Filament table column:

// If empty
        if (empty($record->other)) {
            return '-';
        }

        if (is_array($record->other)) {
            // Example {"cle1":"valeur1"} => ["cle1" => "valeur1"]
            $parts = [];
            foreach ($record->other as $key => $value) {
                if (is_array($value)) {
                    $value = json_encode($value, JSON_UNESCAPED_UNICODE);
                }
                $parts[] = "{$key}: {$value}";
            }
            return implode(', ', $parts);
        }

        return '-';
    })
    ->toggleable(isToggledHiddenByDefault: true),

And in the model I cast the attribute like this:

protected $casts = [
    'other' => 'array',
];

**Comment by emmanpbarrameda** [comment]
**json value:**
[{"deceased_name":"emmans","deceased_birthdate":"2015-12-04","deceased_age":9,"deceased_deathdate":"2024-12-26","deceased_gender":"1"},{"deceased_name":"test test test","deceased_birthdate":"2013-12-20","deceased_age":11,"deceased_deathdate":"2024-12-19","deceased_gender":"2"}]

display json data on table textcolumn:

```
                //?
                Tables\Columns\TextColumn::make('wake_services_repeatable_info')
                    ->label('Deceased Name')
                    ->getStateUsing(function ($record) {
                        if (!$record->wake_services_repeatable_info) return null;
                        
                        return collect($record->wake_services_repeatable_info)
                            ->pluck('deceased_name')
                            ->filter()
                            ->join(', ');
                    })
                    ->searchable(query: function (Builder $query, string $search) {
                        return $query->whereJsonContains('wake_services_repeatable_info', ['deceased_name' => $search]);
                    }),
```

![image](https://github.com/user-attachments/assets/c5af4125-a5a8-4abe-92dd-d916b7339474)
